File: gachisa-backend/src/main/java/com/gachisa/order/service/DeliveryCompletionScheduler.java
package com.gachisa.order.service;

import lombok.RequiredArgsConstructor;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class DeliveryCompletionScheduler {

    private final OrderService orderService;

    @Scheduled(fixedDelayString = "${order.delivery-completion-delay-ms:60000}")
    public void completeDeliveries() {
        orderService.completeDeliveriesDue();
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/order/service/DeliveryCompletionScheduler.kt
package com.gachisa.order.service

import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class DeliveryCompletionScheduler(private val orderService: OrderService) {
    @Scheduled(fixedDelayString = "\${order.delivery-completion-delay-ms:60000}")
    fun completeDeliveries() {
        orderService.completeDeliveriesDue()
    }
}
